lib/dust.js

```diff
@@ -973,25 +995,34 @@
     return this.map(function(chunk) {
       thenable.then(function(data) {
         if (bodies) {
-          chunk = chunk.section(data, context, bodies);
+          chunk.section(data, context, bodies).end();
         } else {
           // Actually a reference. Self-closing sections don't render
-          chunk = chunk.reference(data, context, auto, filters);
+          chunk.reference(data, context, auto, filters).end();
         }
-        chunk.end();
       }, function(err) {
-        var errorBody = bodies && bodies.error;
-        if(errorBody) {
-          chunk.render(errorBody, context.push(err)).end();
-        } else {
-          dust.log('Unhandled promise rejection in `' + context.getTemplateName() + '`', INFO);
-          chunk.end();
-        }
+        dust.log('Unhandled promise rejection in `' + context.getTemplateName() + '`', INFO);
+        chunk.renderError(err, context, bodies).end();
       });
     });
   };
 
   /**
+   * Render an error body if available
+   * @param thenable {Thenable} the target thenable to await
+   * @param context {Context} context to use to render the error
+   * @param bodies {Object} may optionally contain an "error" which will be rendered
+   * @return {Chunk}
+   */
+  Chunk.prototype.renderError = function(err, context, bodies) {
```

Comment:
Thanks.  Should I move it off of the Chunk object?  When I added it originally I wasn't considering that I was updating a public api that's pretty well documented here: http://www.dustjs.com/docs/helper-api/